### DISCUSSION TITLE
Configuration for @parcel/service-worker?

### DISCUSSION THREAD

**Discussion by SoftCreatR** [parcel-bundler/parcel#7122] [Q&A]
Is there a way to configure @parcel/service-worker? I've added a simple sw.js:

```js
import {manifest, version} from '@parcel/service-worker';

async function install() {
    const cache = await caches.open(version)
    await cache.addAll(manifest);
}

addEventListener('install', e => e.waitUntil(install()))

async function activate() {
    const keys = await caches.keys()
    await Promise.all(
        keys.map(key => key !== version && caches.delete(key))
    );
}

addEventListener('activate', e => e.waitUntil(activate()))
```

and parcel does its magic just fine. However, in my case, one file that is being bundled isn't accessible (error page, which is only used and accessible by the webserver), but attempted to be cached.

My workaround is to filter out that file (and all files associated):

```js
import {manifest, version} from '@parcel/service-worker';

async function install() {
    let filteredManifest = []
    const cache = await caches.open(version)

    for (let i = 0; i < manifest.length; i += 1) {
        const index = manifest[i].indexOf("/error")

        if (index === -1) {
            filteredManifest.push(manifest[i])
        }
    }

    await cache.addAll(filteredManifest)
}

addEventListener('install', e => e.waitUntil(install()))

async function activate() {
    const keys = await caches.keys()

    await Promise.all(
        keys.map(key => key !== version && caches.delete(key))
    );
}

addEventListener('activate', e => e.waitUntil(activate()))
```

That works just fine so far, but I wonder if there's a more "native" way to exclude certain files from being processed in the SW build (and maybe some more configuration options that I'm not aware of, like adding files manually to the cache?).

**Comment by grimsteel** [comment]
I needed something like this in my own project so I made a plugin:
https://github.com/grimsteel/parcel-custom-service-worker
